src/transform-applier.js

```diff
@@ -568,8 +574,15 @@ const transformStrokeWidths = function (svgTag, windowRef, bboxForTesting) {
                     }
                 }
 
-                const newRef = _createGradient(gradientId, svgTag, bbox, matrix);
-                if (newRef) fill = newRef;
+                if (fillGradientId) {
+                    const newFillRef = _createGradient(fillGradientId, svgTag, bbox, matrix);
+                    if (newFillRef) fill = newFillRef;
+                }
+
+                if (strokeGradientId) {
+                    const newStrokeRef = _createGradient(strokeGradientId, svgTag, bbox, matrix);
+                    if (newStrokeRef) stroke = newStrokeRef;
+                }
             }
```

Comment:
It seems like fillGradientId and strokeGradientId might be the same, and if they're the same, we should try not to make a duplicate definition (although I'm not sure if it hurts anything if we do have duplicates)